project/ws/app/src/lib/routes/app-toc/components/karmapoints-panel/karmapoints-panel.component.ts

```diff
@@ -1,41 +1,75 @@
 import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core'
 import { TranslateService } from '@ngx-translate/core'
+
 import { MultilingualTranslationsService } from '@sunbird-cb/utils/src/public-api'
+
 @Component({
   selector: 'ws-app-karmapoints-panel',
   templateUrl: './karmapoints-panel.component.html',
   styleUrls: ['./karmapoints-panel.component.scss'],
 })
 export class KarmaPointsPanelComponent implements OnInit {
-    @Input() btntype: any
-    @Input() data: any = []
-    @Input() btnCategory = ''
-    @Input() pCategory = ''
-    @Output() clickClaimKarmaPoints = new EventEmitter<string>()
-    kpData: any
 
   constructor(
     private translate: TranslateService,
-    private langtranslations: MultilingualTranslationsService
+    private langTranslations: MultilingualTranslationsService
   ) {
     if (localStorage.getItem('websiteLanguage')) {
       this.translate.setDefaultLang('en')
       const lang = localStorage.getItem('websiteLanguage')!
       this.translate.use(lang)
     }
   }
+  @Input() btntype: any
+  @Input() data: any = []
+  @Input() btnCategory = ''
+  @Input() pCategory = ''
+  @Output() clickClaimKarmaPoints = new EventEmitter<string>()
+  kpData: any
+
+  karmaPointsSlider: any
 
   ngOnInit() {
     this.data.forEach((item: any) => {
-        if (item.displayButton === this.btntype) {
-            this.kpData = item
-        }
+      if (item.displayButton === this.btntype) {
+          this.kpData = item
+      }
     })
+    this.constructNudgeData()
   }
 
   onClickOfClaim() {
     this.clickClaimKarmaPoints.emit('claim')
   }
+  constructNudgeData() {
+    const nudgeData: any = {
+      type: 'karma-points',
+      iconsDisplay: false,
+      cardClass: 'slider-container',
+      height: 'auto',
+      width: '',
+      sliderData: [],
+      negativeDisplay: false,
+      'dot-default': 'dot-grey',
+      'dot-active': 'dot-active',
+    }
+
+    nudgeData.sliderData = [{
```

Comment:
Please confirm this should not be part of code

Reply:
@vishnubansaltarento  I will take that code out once KB-2942 is fixed. Meanwhile you can merge this PR and it won't affect any other functionality